File: cpp-data-vis-temp.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using System.IO;

class Program
{
    static void Main(string[] args)
    {
        string method;
        List<double> x = new List<double>();
        List<double> y = new List<double>();
        List<string> labels = new List<string>();

        Console.WriteLine("Please enter the name of the file to import data from: ");
        string fileName = Console.ReadLine();

        try
        {
            using (StreamReader sr = new StreamReader(fileName))
            {
                Console.WriteLine("Please enter a visualization method (line, scatter, bar, hist, pie): ");
                method = Console.ReadLine();

                if (method == "line" || method == "scatter" || method == "bar")
                {
                    string line;
                    while ((line = sr.ReadLine()) != null)
                    {
                        var values = line.Split(',');
                        x.Add(Convert.ToDouble(values[0]));
                        y.Add(Convert.ToDouble(values[1]));
                    }
                }
                else if (method == "hist" || method == "pie")
                {
                    string line;
                    while ((line = sr.ReadLine()) != null)
                    {
                        x.Add(Convert.ToDouble(line));
                    }
                }
            }
        }
        catch (Exception e)
        {
            Console.WriteLine("Error reading the file: " + e.Message);
        }

        if (method == "pie")
        {
            Console.WriteLine("Enter labels for each data value (separated by spaces): ");
            string label;
            while ((label = Console.ReadLine()) != null)
            {
                labels.Add(label);
            }
        }

        // Visualizing the data
        Visualize(method, x, y, labels);
    }

    static void Visualize(string method, List<double> x, List<double> y, List<string> labels)
    {
        // Checking the selected method
        if (method == "line")
        {
            // Plotting line graph
            // code for plotting line graph
        }
        else if (method == "scatter")
        {
            // Plotting scatter graph
            // code for plotting scatter graph
        }
        else if (method == "bar")
        {
            // Plotting bar graph
            // code for plotting bar graph
        }
        else
			static void Visualize(string method, List<double> x, List<double> y, List<string> labels)
    {
        // Checking the selected method
        if (method == "line")
        {
            // Plotting line graph
            // code for plotting line graph
        }
        else if (method == "scatter")
        {
            // Plotting scatter graph
            // code for plotting scatter graph
        }
        else if (method == "bar")
        {
            // Plotting bar graph
            // code for plotting bar graph
        }
        else if (method == "hist")
        {
            // Plotting histogram
            // code for plotting histogram
        }
        else if (method == "pie")
        {
            // Plotting pie chart
            // code for plotting pie chart
        }
        else
        {
            Console.WriteLine("Invalid visualization method selected.");
        }
    }
}

File: cpp-data-vis-temp.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <numeric>
#include <cmath>
#include <matplotlibcpp.h>

namespace plt = matplotlibcpp;

void visualize(std::string method, std::vector<double> x, std::vector<double> y, std::vector<std::string> labels) {
    if (method == "line") {
        plt::plot(x, y);
    } else if (method == "scatter") {
        plt::scatter(x, y);
    } else if (method == "bar") {
        plt::bar(x, y);
    } else if (method == "hist") {
        plt::hist(x);
    } else if (method == "pie") {
        plt::pie(x, labels);
    } else {
        std::cout << "Invalid visualization method selected." << std::endl;
    }

    plt::show();
}

int main() {
    std::string method;
    std::vector<double> x, y;
    std::vector<std::string> labels;

    std::cout << "Please enter a visualization method (line, scatter, bar, hist, pie): ";
    std::cin >> method;

    if (method == "line" || method == "scatter" || method == "bar") {
        std::cout << "Enter x values (separated by spaces): ";
        for (double value; std::cin >> value;) {
            x.push_back(value);
        }
        std::cin.clear();

        std::cout << "Enter y values (separated by spaces): ";
        for (double value; std::cin >> value;) {
            y.push_back(value);
        }
        std::cin.clear();

    } else if (method == "hist" || method == "pie") {
        std::cout << "Enter data values (separated by spaces): ";
        for (double value; std::cin >> value;) {
            x.push_back(value);
        }
        std::cin.clear();
    }

    if (method == "pie") {
        std::cout << "Enter labels for each data value (separated by spaces): ";
        std::string label;
        while (std::cin >> label) {
            labels.push_back(label);
        }
        std::cin.clear();
    }

    visualize(method, x, y, labels);

    return 0;
}
